lighthouse-core/audits/jankless-font.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * @license Copyright 2020 The Lighthouse Authors. All Rights Reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License. You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions and limitations under the License.
+ */
+'use strict';
+
+/**
+ * @fileoverview
+ * Audit that checks whether fonts that use `font-display: optional` were preloaded.
+ */
+
+const Audit = require('./audit.js');
+const i18n = require('./../lib/i18n/i18n.js');
+const FontDisplay = require('./../audits/font-display.js');
+const PASSING_FONT_DISPLAY_REGEX = /^(optional)$/;
+const NetworkRecords = require('../computed/network-records.js');
+
+const UIStrings = {
+  /** Title of a Lighthouse audit that provides detail on whether fonts that used `font-display: optional` were preloaded. This descriptive title is shown to users when all fonts that used `font-display: optional` were preloaded. */
+  title: 'Fonts with `font-display: optional` are preloaded',
+  /** Title of a Lighthouse audit that provides detail on whether fonts that used `font-display: optional` were preloaded. This descriptive title is shown to users when one or more fonts used `font-display: optional` and were not preloaded. */
+  failureTitle: 'Fonts with `font-display: optional` are not preloaded',
+  /** Description of a Lighthouse audit that tells the user why they should preload fonts if they are using `font-display: optional`. This is displayed after a user expands the section to see more. No character length limits. 'Learn More' becomes link text to additional documentation. */
+  description: 'Preloading optional fonts can reduce layout shifts and improve CLS. [Learn More](https://web.dev/optimize-cls/#web-fonts-causing-foutfoit)',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+class JanklessFontAudit extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'jankless-font',
```

Comment:
layout shifting used to be called "layout jank".. and the webperf api was initially called the layout jank API but it was renamed. I was one of a few people that thought that "jank" isn't a good term to use for layout shifting. 

in general the term "jank" in webperf refers to lag/hitching that affects animation smoothness. 

https://jakearchibald.github.io/jank-invaders/ 👾 👾 👾 👾 

---------

we've kinda learned that its nice when our ID has similar to the audit's title/failureTitle, so we'll want to use similar keywords.

... 

all this said. this audit pulled me deep into a rabbit hole this afternoon so I raised some questions about the scope back in the issue: https://github.com/GoogleChrome/lighthouse/issues/11227#issuecomment-673745102

Reply:
Let's go with `preload-fonts` as the recommendation is about preloading fonts to get them rendered sooner. (Less shifting is just a side effect)  We all learned this subtle distinction after digging in here. (TBH i could see us updating the existing web.dev docs to make this more clear)

Reply:
Sounds good, I'll change the name to `preload-fonts`!